File: prototype/Assets/microcosmicWar/Scripts/zz/zzTimer.js

var interval:float;

protected var frequencyTimer=zzFrequencyTimer();

function Start()
{
	setInterval(interval);
}

function setImpFunction(pFunc)
{
	frequencyTimer.setImpFunction(pFunc);
}

function setInterval( pInterval: float )
{
	//print(1.0/pInterval);
	//防止在 Start 前执行,使interval未赋值执行setInterval
	interval = pInterval;
	frequencyTimer.setFrequencyOfImp(1.0/pInterval);
}

function Update ()
{
	frequencyTimer.Update ();
}

File: prototype/Assets/microcosmicWar/Scripts/zz/zzTimer.cs

using UnityEngine;
using System.Collections;

public class zzTimer : MonoBehaviour
{

    [SerializeField]
    public float interval;

    public zzTimerClass timer = new zzTimerClass();

    public float timePos
    {
        get { return timer.timePos; }
        set { timer.timePos = value; }
    }

    void Start()
    {
        setInterval(interval);
    }

    public void setImpFunction(zzUtilities.voidFunction pFunc)
    {
        timer.setImpFunction(pFunc);
    }

    public void addImpFunction(zzUtilities.voidFunction pFunc)
    {
        timer.addImpFunction(pFunc);
    }

    public void setInterval(float pInterval)
    {
        interval = pInterval;
        timer.setInterval(pInterval);
    }

    void Update()
    {
        timer.Update();
    }
}